src/main/java/com/spikes2212/dashboard/AutoChooser.java

```diff
@@ -0,0 +1,189 @@
+package com.spikes2212.dashboard;
+
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * A class that extends {@link SendableChooser} in order to make an autonomous command chooser.
+ *
+ * @author Ofri Rosenbaum
+ * @see SendableChooser
+ */
+public class AutoChooser extends SendableChooser<Command> {
+
+    /**
+     * The default name that will be given to the {@link Namespace} in case no name was given in the constructor.
+     */
+    public static final String DEFAULT_NAMESPACE_NAME = "autonomous chooser";
+
+    /**
+     * The incrementation value that will be passed when
+     * using {@link AutoChooser#AutoChooser(Command, String, Object...)}.
+     */
+    private static final int INCREMENTATION_VALUE_ONE = 3;
+
+    /**
+     * The incrementation value that will be passed when
+     * using {@link AutoChooser#AutoChooser(Namespace, Command, String, Object...)}.
+     */
+    private static final int INCREMENTATION_VALUE_TWO = 4;
+
+    /**
+     * The {@link Namespace} that this {@link AutoChooser} will be on.
+     */
+    protected final Namespace namespace;
+
+    /**
+     * A {@link HashSet} that contains the names of all the options in this {@link AutoChooser}.
+     */
+    private final HashSet<String> names = new HashSet<>();
+
+    /**
+     * Creates an {@link AutoChooser} from the given commands, where the first command will be the default option and the
+     * rest of the commands will be the rest of the options.
+     *
+     * <p>All options will have names matching their command's {@link Command#getName()}. In case of multiple
+     * instances of the same class, numbers will also be added to differentiate between the names.</p>
+     *
+     * @param namespace the namespace that this {@link AutoChooser} will be on
+     * @param defaultOption the default command this {@link AutoChooser} will use as the default option
+     * @param options the commands that will be added to this {@link AutoChooser} other than the default command
+     */
+
+    public AutoChooser(Namespace namespace, Command defaultOption, Command... options) {
+        this.namespace = namespace;
+        String defaultName = defaultOption.getName();
+        addName(defaultName);
+        setDefaultOption(defaultName, defaultOption);
+        for (Command option : options) {
```

Comment:
I personally prefer `forEach` but you do you

Reply:
Seems unnecessary